### DISCUSSION TITLE
Load CefSettings from file ?

### DISCUSSION THREAD

**Discussion by barts2108** [cefsharp/CefSharp#3367] [Q&A]
I am up to integrating CefSharp.WinForms into a scada package that uses basicscript. I want the browser background to be black and in a VS (C#) project I can achieve that by setting CefSettings() Background Color to 0xFF000000. I also read that this can be done using command line arguments. That makes me think that the CefSettings (or command line arguments) are used by the CefSharp.BrowserSubprocess.exe. Is that thought correct ?

Since I only added the control, I do not have CefSettings and because the scada is using BasicScript, I might have a hard time getting the subprocress started with the black background color.

Is there an option where the CefSettings() are read from file ? If so, how should that be done ? If not, is it possible to add such feature ?

**Comment by amaitland** [comment]
> I am up to integrating CefSharp.WinForms into a scada package that uses basicscript.

Never heard of it. 

> I also read that this can be done using command line arguments.

I don't believe that is correct.

> That makes me think that the CefSettings (or command line arguments) are used by the CefSharp.BrowserSubprocess.exe. Is that thought correct ?

Each property will potentially be handled differently. Some are `CEF` specific, some are generic to `Chromium`. I doubt that `Background Color` is passed to the render process via command line args.

> Since I only added the control, I do not have CefSettings and because the scada is using BasicScript, I might have a hard time getting the subprocress started with the black background color.

You can use [BrowserSettings.BackgroundColor](http://cefsharp.github.io/api/86.0.x/html/P_CefSharp_BrowserSettings_BackgroundColor.htm) to modify the settings when you create a `ChromiumWebBrowser` instance.

> Is there an option where the CefSettings() are read from file ?

No

> If so, how should that be done ?

In general when integrating into a `3rd party` framework creating a wrapper library of your own to handle the implementation detail is probably the best approach. Plenty of others have done this one example being https://github.com/javis86/cefinteropcontrol (I've not suggesting you need to create anything so elaborate, just to give you some pointers).

**Reply by barts2108** [reply]
> Never heard of it.

Be happy about that. I found it is BasicScript version 2.25 dating back to 1996, identifying my i7-6core cpu as a pentium pro/II/III
In a company that claims innovative, using "stone-age" technology does not make me happy. Unfortunately I am tied to that 😭 
[(https://www.google.com/search?q=basicscript+2.25)](https://www.google.com/search?q=basicscript+2.25)

> You can use BrowserSettings.BackgroundColor to modify the settings when you create a ChromiumWebBrowser instance.

@amaitland : While typing an extensive reply on your remarks, I tested this in my Visual Studio test project, and it looks likt this is doing the same as the CefSettings.BackgroundColor. Thanks for that, I will test if I can manage to get it working to get some modern browser control running with 25 year old scripting lanuage 😞
